### DISCUSSION TITLE
PR/branch limit per package.json

### DISCUSSION THREAD

**Discussion by namrog84** [renovatebot/renovate#12632] [Request Help] [answered]
We have a monorepo with 20+ different sets of package.json  (no yarn workspaces or anything) just all in independent folders, all managed by different teams, completely separate things.

I tried doing a thing in the packageRules
```
   {
      "matchPaths": ["package1/package.json"],
      "prConcurrentLimit": 3,
      "branchPrefix":"renovate/package1/",
    },
    {
      "matchPaths": ["package2/package.json"],
      "prConcurrentLimit": 3,
      "branchPrefix":"renovate/package2/",
    },
```

With a global limit of like
"branchConcurrentLimit": 10,
"prConcurrentLimit": 10,

I want to limit it to 2-3 per package, but also have no more than 10 PR/branches out at any given time. 

The documentation seems to say "Note that this limit is enforced on a per-repository basis"

1. Is there any way I can set it up a limit of PRs/branches on a per package basis? 
2. Alternatively, is there any (hackish?) way around the 1 renovate.json per repo?  This makes large monorepo kinda difficult to manage because we a lot are managed by different teams/needs.  Note: This is self hosted ADO, so not sure if there is a way to somehow hot-swap them out or something either for different pipelines updating different packages inside the monorepo?

**Comment by HonkingGoose** [comment, marked as answer]
I think the following feature requests might improve things for you once they're implemented:

- https://github.com/renovatebot/renovate/issues/10546
- https://github.com/renovatebot/renovate/issues/6684
- https://github.com/renovatebot/renovate/issues/5825

Can you take a look and see if any of these features would help you?

I recommend you take a look at our "Noise Reduction" [^noise] documentation, and see if any of the tips in there could be helpful for you.

[^noise]: https://docs.renovatebot.com/noise-reduction/

**Reply by namrog84** [reply]
Yep that's exactly what I was looking for, totally missed those when I was searching!  I hope they come one day soonish!
